File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int GetGCD(int a, int b){
    if(a % b == 0) return b;
    else return GetGCD(b, a % b);
}

vector<int> solution(int n, int m) {
    int gcd = GetGCD(n, m);
    int lcm = (n * m) / gcd; 
    return {gcd, lcm};
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    private int GetGCD(int a, int b){
        return a % b == 0 ? b : GetGCD(b, a % b);
    }
    
    public int[] solution(int n, int m) {
        int gcd = GetGCD(n, m);
        int lcm = (n * m) / gcd; 
        return new int[]{gcd, lcm};
    }
}